### PR TITLE
state: Get correct locked table in StateMarketParticipants

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: node/impl/full/state.go [hunk 1]
```diff
@@ -489,13 +489,11 @@ func (a *StateAPI) StateMarketDeals(ctx context.Context, tsk types.TipSetKey) (m
 
 		var s market.DealState
 		if err := sa.Get(ctx, i, &s); err != nil {
-			if err != nil {
```
Comment (magik6k) [inline node/impl/full/state.go:492]: Not related to the PR title, but this was always true

File: node/impl/full/state.go [hunk 2]
```diff
@@ -489,13 +489,11 @@ func (a *StateAPI) StateMarketDeals(ctx context.Context, tsk types.TipSetKey) (m
...
-				if _, ok := err.(*amt.ErrNotFound); !ok {
-					return xerrors.Errorf("failed to get state for deal in proposals array: %w", err)
-				}
-
-				s.SectorStartEpoch = -1
+			if _, ok := err.(*amt.ErrNotFound); !ok {
```
Comment (Kubuxu) [inline node/impl/full/state.go:492]: Use: `errors.As`
```
var notFound *amt.ErrNotFound
if errors.As(err, notFound) {

}
```

Comment (magik6k) [reply node/impl/full/state.go:492]: but `if _, ok := err.(*amt.ErrNotFound); !ok {` is one line :c (and arguably more readable)

Comment (Kubuxu) [reply node/impl/full/state.go:492]: It will break if anything along the call chain uses `errors.Errorf("bla bla bla: %w")`

**Pull request merged**